c/unixFileService.c

```diff
@@ -920,6 +920,38 @@ static int serveUnixFileMetadata(HttpService *service, HttpResponse *response) {
   return 0;
 }
 
+static int serveUnixFileChangeOwner (HttpService *service, HttpResponse *response) {
+  HttpRequest *request = response->request;
+  char *routeFileFrag = stringListPrint(request->parsedFile, 2, 1000, "/", 0);
```

Comment:
Can we check the input parameters? For example, `directoryChangeOwnerAndRespond` function uses `recursive` as is, without checking for NULL. Currently it's not clear which parameters are optional.

Reply:
The code uses strcmp to set the bool recursive.  strcmp will return false if the string is null.  We can add an additional test for null to make it more clear.

Reply:
> strcmp will return false if the string is null. 

It may work or it may segfault/S0C4, I think it's implementation defined.

Reply:
On z/OS strcmp() will just invoke a builtin which boils down to a CLST instruction with no check for NULL. I suspect it will result in a 0C4.

Reply:
FWIW, I don't use strcmp! I prefer to use macros or inline functions that check for null. 

```
inline bool streq(const char * lhs, const char * rhs) {            
  return lhs != NULL && rhs != NULL && strcmp(lhs, rhs) == 0;      
}                                                                  
```

Reply:
I have put in the additional test for NULL before the strcmp so there shouldn't be a problem.

Reply:
I see this here now: https://github.com/zowe/zowe-common-c/pull/104/files#diff-5a6b6901a39402924b29b573b54bbc0eR571